cellpack/bin/simularium_converter.py

```diff
@@ -282,67 +324,71 @@ def get_euler(self, data_in):
-            self.display_data[ingredient_name] = DisplayData(
-                name=ingredient_name,
+    def loop_through_ingredients(
+        self,
+        result_compartments_ingredients,
+        recipe_compartments_ingredients,
+        time_step_index,
+    ):
+        for ingredient_key in recipe_compartments_ingredients:
+            if ingredient_key not in result_compartments_ingredients:
+                continue
+            ingredient_data = recipe_compartments_ingredients[ingredient_key]
+            ingredient_results_data = result_compartments_ingredients[ingredient_key]
+            display_data = self.get_ingredient_display(ingredient_data)
+            self.display_data[ingredient_key] = DisplayData(
+                name=ingredient_key,
                 display_type=display_data["display_type"],
                 url=display_data["url"],
             )
-            if cytoplasm_data is None and container_data is None:
-                continue
-            if cytoplasm_data is not None:
-                data = cytoplasm_data
-                if len(cytoplasm_data["results"]) > 0:
-                    for j in range(len(data["results"])):
-                        self.unpack_positions(
-                            data, time_step_index, ingredient_name, j, agent_id
-                        )
-                        agent_id = agent_id + 1
-
-                elif cytoplasm_data["nbCurve"] > 0:
-                    for i in range(data["nbCurve"]):
-                        self.unpack_curve(
-                            data, time_step_index, ingredient_name, i, agent_id
-                        )
-                        agent_id = agent_id + 1
-            if container_data is not None:
-                data = container_data
-                comp_id = container_data["compNum"]
-                if len(data["results"]) > 0:
-                    for j in range(len(data["results"])):
-                        self.unpack_positions(
-                            data, time_step_index, ingredient_name, j, agent_id, comp_id
-                        )
-                        agent_id = agent_id + 1
-
-                elif data["nbCurve"] > 0:
-                    for i in range(data["nbCurve"]):
-                        self.unpack_curve(
-                            data, time_step_index, ingredient_name, i, agent_id
-                        )
-                        agent_id = agent_id + 1
+            if len(ingredient_results_data["results"]) > 0:
+                for j in range(len(ingredient_results_data["results"])):
+                    self.unpack_positions(
+                        ingredient_results_data,
+                        time_step_index,
+                        ingredient_key,
+                        j,
+                        self.agent_id_counter,
+                    )
+                    self.agent_id_counter = self.agent_id_counter + 1
+            elif ingredient_results_data["nbCurve"] > 1000000:
```

Comment:
this doesn't seem right, why greater than `1000000`?

Reply:
Sorry I didnt wanted to debug the curve and wanted to skipped them. I put a TODO comment.